examples/nh_mountain.py

```diff
@@ -23,7 +23,12 @@
 H = Constant(H)
 a = Constant(1000.)
 xc = Constant(L/2.)
-new_coords = Function(Vc).interpolate(as_vector([x[0], x[1]+(H-x[1])*a**2/(H*((x[0]-xc)**2+a**2))]))
+smooth_z = True
+if smooth_z:
+    xexpr = Expression(("x[0]","x[1] < zh ? x[1]+pow(cos(0.5*pi*x[1]/zh),6)*h*pow(a,2)/(pow(x[0]-xc,2)+pow(a,2)) : x[1]"), zh=5000., h=1., a=a, xc=xc)
```

Comment:
yes, Andrew told me about this uff conditional yesterday - awesome! I will make a separate issue and convert all the Expressions to ufl interpolations - they appear all over the place at the moment.